<a href="https://colab.research.google.com/github/p09323028/2021s_NTU_Econometrics_II/blob/main/Homework/PS2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **經濟實證與預測二_ Homework 2**
**Link:** https://tinyurl.com/ch77ndts

**吳金擇, P09323028**

**林婧筠, P09323026**

**洪靜玟, P09323019**

**林冠伭, P08323019**

# **Exercise 13.6**
Suppose there are panel data for $T = 2$ time periods for a randomized controlled
experiment, where the first observation $(t = 1)$ is taken before the
experiment and the second observation $(t = 2)$ is for the posttreatment
period. Suppose the treatment is binary; that is, suppose $X_{it} = 1$ if the $i^{th}$ individual
is in the treatment group and $t = 2$
, and $X_{it} = 0$ otherwise. Further
suppose the treatment effect can be modeled using the specification

$$Y_{it} = \alpha_i + \beta_1X_{it} + u_{it}.$$

where $\alpha_i$ are individual-specific effects with a mean of $0$ and a variance of $\sigma^2_{\alpha}$
and $u_{it}$ is an error term, where $u_{it}$ is homoskedastic, $cov(u_{i1}, u_{i2}) = 0$, and
$cov(u_{it}, \alpha_i) = 0$ for all $i$. Let $\widehat{\beta}^{differences}$ denote the differences estimator—
that is, the OLS estimator in a regression of $Y_{i2}$ on $X_{i2}$ with an intercept—
and let $\widehat{\beta}^{diffs-in-diffs}_1$ denote the differences-in-differences estimator—that is, the estimator of $\beta_1$ based on the OLS regression of $\Delta Y_i = Y_{i2} - Y_{i1}$ against
$\Delta X_i = X_{i2} - X_{i1}$ and an intercept.
1. Show that $n Var(\widehat{\beta}^{differences}) \rightarrow (\sigma^2_u + \sigma^2_\alpha) / Var(X_{i2})$. (Hint: Use the
homoskedasticity-only formulas for the variance of the OLS estimator in
Appendix 5.1.)
2. Show that $n Var (\widehat{\beta}^{diffs-in-diffs}_1) \rightarrow 2 \sigma^2_u / Var(X_{i2})$. (Hint: Note that
$X_{i2} - X_{i1} = X_{i2}$. Why?)
3. Based on your answers to (a) and (b), when would you prefer the
differences-in-differences estimator over the differences estimator, based
purely on efficiency considerations?
---

## **13.6-1**
Regression of $Y_{i2}$ on $X_{i2}$ with an intercept:

$Y_{i2} = b_0 + \beta^{differences} X_{i2} + (\alpha_i + u_{i2} -b_0)$


$n Var(\widehat{\beta}^{differences}) = \frac{Var(\alpha_i + u_{i2} -b_0)}{Var(X_{i2})}$

$= \frac{Var(\alpha_i) + Var(u_{i2}) - 2Cov(\alpha_i, u_{i2})}{Var(X_{i2})}$

$= \frac{Var(\alpha_i) + Var(u_{i2}) - 2 \times 0}{Var(X_{i2})}$

$= \frac{\sigma^2_u + \sigma^2_\alpha}{Var(X_{i2})}$

## **13.6-2**
$Y_{i2} - Y_{i1}= \beta_1^{diffs-in-diffs} (X_{i2}-X_{i1}) + (u_{i2}-u_{i1})$

$n Var(\widehat{\beta}^{diffs-in-diffs}) = \frac{Var(u_{i2}-u_{i1})}{Var(X_{i2}-X_{i1})}$

$= \frac{Var(u_{i2}) + Var(u_{i1}) - 2Cov(u_{i2}, u_{i1})}{Var(X_{i2})} \because X_{i1} = 0,  \text{for all } i.$

$= \frac{Var(u_{i2}) + Var(u_{i1}) - 2 \times 0}{Var(X_{i2})}$

$= \frac{2 \times \sigma^2_u}{Var(X_{i2})}$


## **13.6-3**
若 $\sigma_{u}^2 < \sigma_{\alpha}^2$ (individual-specific effects 變異大), 則  differences-in-differences estimator 較優(More efficient)。

# **Exercise 13.7**
Suppose you have panel data from an experiment with $T = 2$ periods (so
$t = 1, 2$). Consider the panel data regression model with fixed individual and
time effects and individual characteristics $W_i$ that do not change over time. Let
the treatment be binary, so that $X_{it} = 1$ for $t = 2$ for the individuals in the treatment
group and $X_{it} = 0$ otherwise. Consider the population regression model

$$Y_{it} = \alpha_i + \beta_1X_{it} + \beta_2(D_i \times W_i) + \beta_0D_t + v_{it}$$

where $\alpha_i$ are individual fixed effects, $D_t$ is the binary variable that equals $1$
if $t = 2$ and equals $0$ if $t = 1$, $D_t \times W_i$ is the product of $D_t$ and $W_i$, and the
$\alpha$’s and $\beta$’s are unknown coefficients. Let $\Delta Y_i = Y_{i2} - Y_{i1}$. Derive Equation
(13.6) (in the case of a single $W$ regressor, so $r = 1$) from this population
regression model.

---



$\left\{ \begin{array}{ll}
Y_{1i} = \alpha_i + \beta_1 X_{i1} + v_{i1}  \\
Y_{2i} = \alpha_i + \beta_1 X_{i2} + \beta_2 W_{i} + \beta_0 + v_{i2}
\end{array} \right.$

$\rightarrow Y_{2i} - Y_{1i} = \beta_1 (X_{i2}-X_{i1}) + \beta_2 W_{i} + \beta_0 + (v_{i2}-v_{i1})$

$\rightarrow \Delta Y_i = \beta_1 \Delta X_i + \beta_2 W_{i} + \beta_0 + \Delta v_i = \beta_0 + \beta_1 X_i + \beta_2 W_{i} + u_i$

# **Exercise 13.10**
Consider the regression model with heterogeneous regression coefficients
$$Y_i = \beta_0 + \beta_{1i}X_i + v_i$$
where $(v_i, X_i, \beta_{1i})$ are $i.i.d$. random variables with $\beta_1 = E(\beta_{1i})$.
1. Show that the model can be written as $Y_i = \beta_0 + \beta_1X_i + u_i$, where $u_i = (\beta_{1i} - \beta_1)X_i + v_i$.
2. Suppose $X_i$ is randomly assigned, so that $E[\beta_{1i}|X_i] = \beta_1$ and $E[v_i|X_i] = 0$. Show that $E[u_i|X_i] = 0$.
3. Show that assumption 1 and assumption 2 of Key Concept 4.3 are satisfied.
4. Suppose outliers are rare, so that $(u_i, X_i)$ have finite fourth moments. Is
it appropriate to use OLS and the methods of Chapters 4 and 5 to estimate
and carry out inference about the average values of $\beta_{0i}$ and $\beta_{1i}$?
5. Now suppose $X_i$ is not randomly assigned, that $E[v_i|X_i] = 0$, but that
$\beta_{1i}$ and $X_i$ are positively correlated, so that observations with largerthan- average values of $X_i$ tend to have larger-than-average values of
$\beta_{1i}$. Are the assumptions in Key Concept 4.3 satisfied? If not, which
assumption(s) is (are) violated? Will the OLS estimator of $\beta_1$ be
unbiased for $E(\beta_{1i})$?

---




## **13.10-1**
$\begin{eqnarray} 
Y_i &=& \beta_0 + \beta_{1i} X_i + v_i   \\
~&=& \beta_0 + \beta_{1i} X_i + v_i + (\beta_1 X_i - \beta_1 X_i) \\
~&=& \beta_0 + \beta_1 X_i + v_i + (\beta_{1i} X_i - \beta_1 X_i) \\
~&=& \beta_0 + \beta_1 X_i + u_i
\end{eqnarray}$

## **13.10-2**
$\begin{eqnarray} 
E[u_i|X_i] &=& E[(\beta_{1i} - \beta_1 )X_i + v_i|X_i]  \\
~&=& X_i E[\beta_{1i}|X_i] - X_i E[\beta_{1}|X_i] + E[v_i|X_i]  \\
~&=& \beta_1 - \beta_1 + 0 \\
~&=& 0
\end{eqnarray}$

## **13.10-3**
由 **13.10-2**, 知 assumption 1 成立；

由 $(v_i, X_i, \beta_{1i})$ are $i.i.d$. random variables, 知 assumption 2 成立。

## **13.10-4**
Outliers are rare $\Rightarrow$ 滿足 Key concept 4.3 assumption 3,

配合 **13.10-3**, 知 it is appropriate to use OLS to estimate and carry out inference.

## **13.10-5**
$Y_i = \beta_0 + \beta_{1i}X_i + v_i$

$\begin{eqnarray}
Cov(\beta_{1i}, X_i) &=& E[(\beta_{1i}-E(\beta_{1i}))(X_{i}-E(X_{i}))] \\
&=& E[(\beta_{1i}-\beta_1)(X_{i}-E(X_{i}))] \\
&=& E[(\beta_{1i}-\beta_1)X_{i}-(\beta_{1i}-\beta_1)E(X_{i})] \\
&=& E[(\beta_{1i}-\beta_1)X_{i}] > 0
\end{eqnarray}$

$\Rightarrow E \{ E[(\beta_{1i}-\beta_1)X_{i}|X_i] \} > 0$

$\Rightarrow E[(\beta_{1i}-\beta_1)X_{i}|X_i] > 0$

知 $E[u_i|X_i] = E[(\beta_{1i}-\beta_1)X_{i}+v_i|X_i] \neq 0$,

Key concept 4.3 assumption 1 不成立,

the OLS estimator of $\beta_1$ would not be unbiased for $E(\beta_{1i})$.

# **Exercise 13.12**
Consider the potential outcomes framework from Appendix 13.3. Suppose $X_i$
is a binary treatment that is independent of the potential outcomes $Y_i(1)$ and
$Y_i(0)$. Let $TE_i = Y_i(1) - Y_i(0)$ denote the treatment effect for individual $i$.
1. Can you consistently estimate $E[Y_i(1)]$ and $E[Y_i(0)]$? If yes, explain
how; if not, explain why not.
2. Can you consistently estimate $E(TE_i)$? If yes, explain how; if not,
explain why not.
3. Can you consistently estimate $Var[Y_i(1)]$ and $Var[Y_i(0)]$? If yes, explain how; if not, explain why not.
4. Can you consistently estimate $Var(TE_i)$? If yes, explain how; if not,
explain why not.
5. Do you think you can consistently estimate the median treatment effect
in the population? Explain.

---



由 Appedix 13.3 及題旨, 知

1. $Y_i = Y_i(1)X_i + Y_i(0)(1-X_i)$

2. $X_i$
is a binary treatment that is independent of the potential outcomes $Y_i(1)$ and
$Y_i(0)$

## **13.12-1**
$\begin{eqnarray} 
E[Y_i|X_i=1] &=& E[Y_i(1)X_i + Y_i(0)(1-X_i)|X_i=1]  \\
~&=& E[Y_i(1)|X_i=1] \\
~&=& E[Y_i(1)]  \text{(因兩者獨立)}\\
\end{eqnarray}$

$\Rightarrow E[Y_i(1)]$ 可由實驗組的樣本平均值估算。

同理可證, $E[Y_i(0)]$ 可由對照組的樣本平均值估算。

## **13.12-2**
$E(TE_i) = E[Y_i(1) - Y_i(0)]$

由 **13.12-1** 知 $E(TE_i)$ 可由實驗組及對照組的樣本平均值之差估算。

## **13.12-3**
$\begin{eqnarray}
Var(Y_i|X_i=1) &=& E(Y_i^2|X_i=1) - [E(Y_i|X_i)=1]^2 \\
&=& E[Y_i(1)^2] - [E(Y_i(1)]^2 \\
&=& Var[Y_i(1)]
\end{eqnarray}$

$\Rightarrow Var[Y_i(1)]$ 可由實驗組的樣本變異數估算。

同理可證, $Var[Y_i(0)]$ 可由對照組的樣本變異數估算。

## **13.12-4**
$Var(TE_i) = Var[Y_i(1)] + Var[Y_i(0)] - 2 Cov[Y_i(1), Y_i(0)]$

$Var[Y_i(1)]$ 及 $Var[Y_i(0)]$ 可由 **13.12-3** 估算, 

但因同一觀測樣本無法同時於實驗組及對照組, 故無法估算 $Cov[Y_i(1), Y_i(0)]$,

致 $Var(TE_i)$ 無法估算。

## **13.12-5**
中位數運算特性與 **13.12-2** 平均數的運算特性不同,

$median[Y_i(1)-Y_i(0)] \neq median[Y_i(1)] - median[Y_i(0)]$

故應無法估算 $median[Y_i(1)-Y_i(0)]$。



# **E13.1**
A prospective employer receives two resumes: a resume from a white job applicant
and a similar resume from an African American applicant. Is the employer
more likely to call back the white applicant to arrange an interview? Marianne
Bertrand and Sendhil Mullainathan carried out a randomized controlled
experiment to answer this question. Because race is not typically included on
a resume, they differentiated resumes on the basis of “white-sounding names”
(such as Emily Walsh or Gregory Baker) and “African American–sounding
names” (such as Lakisha Washington or Jamal Jones). A large collection of fictitious
resumes was created, and the presupposed “race” (based on the “sound”
of the name) was randomly assigned to each resume. These resumes were sent
to prospective employers to see which resumes generated a phone call (a callback)
from the prospective employer. Data from the experiment and a detailed
data description are on the text website, **http://www.pearsonglobaleditions.com**, in the files **Names and Names_Description**.
1. Define the $callback$ rate as the fraction of resumes that generate a phone
call from the prospective employer. What was the callback rate for
whites? For African Americans? Construct a 95% confidence interval
for the difference in the callback rates. Is the difference statistically significant? Is it large in a real-world sense?
2. Is the African American/white callback rate differential different for
men than for women?
3. What is the difference in callback rates for high-quality versus lowquality
resumes? What is the high-quality/low-quality difference for
white applicants? For African American applicants? Is there a significant
difference in this high-quality/low-quality difference for whites versus
African Americans?
4. The authors of the study claim that race was assigned randomly to the
resumes. Is there any evidence of nonrandom assignment?

---



In [ ]:
import numpy as np
import pandas as pd
import statsmodels.formula.api as smf

In [ ]:
!gdown --id '1b1OD7qauTWdefOjojLRRKe0feYyaWPgx' --output names.xlsx

In [ ]:
names = pd.read_excel('names.xlsx')

In [ ]:
names.head()

「種族」以“white-sounding names”及“African American–sounding names”區分。

## E13.1-a

In [ ]:
reg_13_1_a = smf.ols('call_back~black', data=names)
results_13_1_a = reg_13_1_a.fit(cov_type='HC1', use_t=True)
print(results_13_1_a.summary())

                            OLS Regression Results                            
Dep. Variable:              call_back   R-squared:                       0.003
Model:                            OLS   Adj. R-squared:                  0.003
Method:                 Least Squares   F-statistic:                     16.93
Date:                Sat, 17 Apr 2021   Prob (F-statistic):           3.94e-05
Time:                        15:46:24   Log-Likelihood:                -562.24
No. Observations:                4870   AIC:                             1128.
Df Residuals:                    4868   BIC:                             1141.
Df Model:                           1                                         
Covariance Type:                  HC1                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      0.0965      0.006     16.124      0.0

The callback rate for whites: $9.65\%$

The callback rate for African Americans: $6.45\%$

$95\%$ confidence interval: $[-4.73\%, -1.68\%]$

$\alpha = 0.01, Pvalue=0.00 < \alpha \Rightarrow $ 統計上, 種族對回覆率有顯著影響。

因兩者回覆率皆很低, 但因種族就占 $3.20\%$ 之差, 故差異滿大的。

## E13.1-b

In [ ]:
reg_13_1_b = smf.ols('call_back~black+female+black*female', data=names)
results_13_1_b = reg_13_1_b.fit(cov_type='HC1', use_t=True)
print(results_13_1_b.summary())

                            OLS Regression Results                            
Dep. Variable:              call_back   R-squared:                       0.004
Model:                            OLS   Adj. R-squared:                  0.003
Method:                 Least Squares   F-statistic:                     5.995
Date:                Sat, 17 Apr 2021   Prob (F-statistic):           0.000450
Time:                        15:30:44   Log-Likelihood:                -561.75
No. Observations:                4870   AIC:                             1131.
Df Residuals:                    4866   BIC:                             1157.
Df Model:                           3                                         
Covariance Type:                  HC1                                         
                   coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------
Intercept        0.0887      0.012      7.478   

$\alpha = 0.1, black \times female$ 之 $Pvalue = 0.9 > \alpha$

$\Rightarrow$ 統計上, 種族與性別的交互作用不顯著。

## E13.1-c

In [ ]:
reg = smf.ols('call_back~high', data=names)
results = reg.fit(cov_type='HC1', use_t=True)
print(results.t_test('high=0').summary(xname=['high']))

                             Test for Constraints                             
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
high           0.0141      0.008      1.804      0.071      -0.001       0.029


高品質履歷與低品質履歷的回覆率，平均差 $1.41\%$

In [ ]:
reg_13_1_c = smf.ols('call_back~black+high+black*high', data=names)
results_13_1_c = reg_13_1_c.fit(cov_type='HC1', use_t=True)
print(results_13_1_c.summary())

                            OLS Regression Results                            
Dep. Variable:              call_back   R-squared:                       0.004
Model:                            OLS   Adj. R-squared:                  0.004
Method:                 Least Squares   F-statistic:                     6.613
Date:                Sat, 17 Apr 2021   Prob (F-statistic):           0.000187
Time:                        15:37:14   Log-Likelihood:                -559.96
No. Observations:                4870   AIC:                             1128.
Df Residuals:                    4866   BIC:                             1154.
Df Model:                           3                                         
Covariance Type:                  HC1                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      0.0850      0.008     10.605      0.0

The high-quality/low-quality difference for white applicants: $2.29\%$

The high-quality/low-quality difference for white applicants: $\widehat{\beta}_{high} + \widehat{\beta}_{black \times high} = 0.52\%$

$\alpha = 0.1, black \times high$ 之 $Pvalue = 0.25 > \alpha$

$\Rightarrow$ 統計上, 種族與履歷品質的交互作用不顯著。

## E13.1-d

In [ ]:
reg_13_1_d = smf.ols('black~ofjobs+yearsexp+honors+volunteer+military+empholes+workinschool+email+computerskills+specialskills+college', data=names)
results_13_1_d = reg_13_1_d.fit(cov_type='HC1', use_t=True)
print(results_13_1_d.summary())

                            OLS Regression Results                            
Dep. Variable:                  black   R-squared:                       0.002
Model:                            OLS   Adj. R-squared:                 -0.001
Method:                 Least Squares   F-statistic:                    0.6713
Date:                Sat, 17 Apr 2021   Prob (F-statistic):              0.767
Time:                        16:10:42   Log-Likelihood:                -3530.9
No. Observations:                4870   AIC:                             7086.
Df Residuals:                    4858   BIC:                             7164.
Df Model:                          11                                         
Covariance Type:                  HC1                                         
                     coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------
Intercept          0.4683      0.033     14.

將 $black$ 與 履歷相關變數 做迴歸分析, 

由於迴歸式整體 $Pvalue=0.77$, 故統計上, 無證據顯示種族為非隨機指定至履歷。

(與種族較相關的履歷變數為 $computerskills$)